framework-cloud/cloud-deployment-plugin/src/main/java/org/kie/cloud/plugin/Undeploy.java

```diff
@@ -38,18 +40,19 @@
     private MavenProject mavenProject;
 
     @Override public void execute() throws MojoExecutionException, MojoFailureException {
-        DeploymentScenarioBuilderFactory deploymentScenarioBuilderFactory = DeploymentScenarioBuilderFactory.getInstance();
-        deploymentScenarioBuilderFactory.deleteNamespace(getNamespace());
+        DeploymentScenarioBuilderFactory deploymentScenarioBuilderFactory = DeploymentScenarioBuilderFactoryLoader
+                .getInstance(loadCloudProperties().getProperty(CLOUD_API_IMPLEMENTATION_PROPERTY));
+        deploymentScenarioBuilderFactory.deleteNamespace(loadCloudProperties().getProperty(NAMESPACE_PROPERTY));
     }
 
-    private String getNamespace() {
+    private Properties loadCloudProperties() {
```

Comment:
I don't like that instead of domain specific properties (like namespace and now the newly added cloud api implementation property) we're now returning the entire properties.

Wouldn't it be better to encapsulate the properties in some more domain specific class, to let the api user know what properties are retrievable? Otherwise api user will have to look at usages of this method and search what is being stored into it and then also look for constants representing valid keys etc.